### DISCUSSION TITLE
Welcome to KServe Discussions!

### DISCUSSION THREAD

**Discussion by yuzisun** [kubeflow/kfserving#1801] [Q&A]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by brness** [comment]
Hello everyone, I'm a newbee to kfserving. And I just wonder how to traffic flow into the right pod. As far as I know, the traffic first go through the gateway (In my situation, It's the _knative-local-gateway_) and according to the vs, the traffic goes to the _knative-local-gateway.istio-system.svc.cluster.local_ with a header of _sklearn-iris-cy-predictor-default.kubeflow-user-example-com.svc.cluster.local_. and somehow the traffic go to the pod of modelserving. Can someone explain it ? Because this is quite confused for me and I install the kiali but the draw of the flow didn't work, and I'll be appreciated if somebody can show me the traffic route and vs and gateway with a simple graph.

**Reply by theofpa** [reply]
Hello @brness and welcome to kfserving! You can find troubleshooting information as well as the diagram which explains the flow you ask in [this guide](https://github.com/kubeflow/kfserving/blob/master/docs/KFSERVING_DEBUG_GUIDE.md).

**Reply by brness** [reply]
Thank you for your replying@theofpa. And the problem now of mine is hard to match the gateway and vs to trace the traffic of the prediction of the serving model. I did it successfully on the sample of the single model in my k8s, I read the debugging guide and it help me a lot. but i can't fully understand the part of why the traffic already goes into the gateway of istio-ingress-gateway and again go back to the service of knative-local-gateway in the istio-system namespace, why not go and get the prediction po or svc instead, what's the insight of the step? and during my multimodel-serving test, it failed and i just want to check where maybe the wrong point. but i can't debug the real point, that's why i ask about how single model serving traffic flows! sound quite stupid but i try kiali but no flow pictures shows bad luck. Still great to have you reply in such a short time!

**Reply by brness** [reply]
hello @yuzisun , I already figure the flow the traffic out, but during my experiment of multimodel-serving, it somehow can't recieve a response all feedback, the logs shows the request does not come to the container but I don't where maybe the problem, and I already put it on a issue and as for now no one replies yet, can you have a look, thanks a lot!

**Reply by yuzisun** [reply]
@gsunner sorry did not see this thread until now, can you point me the issue you created?

**Comment by DARSHAN13302** [comment]
I’m Darshan.

I’m working as a researcher on an Enterprise LLMOps project and trying to validate real pain points in production LLM systems.

I’m not selling anything. I just want to understand what practitioners actually struggle with so we don’t build a fake classroom solution.

Could I ask you 3 quick questions?

What are the biggest pain points your team faces in production LLM systems?
What tools or workarounds are you using today?
What is still unsolved that you wish a better solution existed for?

Even a 2-line reply would help a lot.
